docs/madr/decisions/074-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,152 @@
+
+Some internal resources do not map clearly to Kuma resources in the data store, making it difficult to relate metrics to Kuma abstractions.
+
+There are also resources that are in a bit of a gray area, like `meshtrace_datadog` which is a cluster created for `MeshTrace` policy.
+It seems internal, but it could be traced down to MeshTrace resource `kri_meshtrace_mesh-1__kuma-system_my-meshtrace_`.
+
+#### Examples of kuma resource names
+
+Listeners:
+- inbound
+  - kuma:envoy:admin
+  - _kuma:dynamicconfig
+  - listener./tmp/kuma-dp-728637052/kuma-mesh-metric-config.sock
+  - listener.0.0.0.0_15001
+  - listener.0.0.0.0_15006
+  - listener.10.42.0.9_9901
+  - listener.[__]_15001 # is it ipv6?
+  - probe_listener
+  - prometheus_listener
+- outbound
+  - outbound:passthrough:ipv4
+  - outbound:passthrough:ipv6
+
+Clusters:
+- access_log_sink
+- kuma:envoy:admin
+- kuma:metrics:hijacker
+- kuma:readiness
+- meshtrace_[zipkin|datadog|otel]
+- envoy_grpc_streams (ads)
+
+Virtual Hosts:
+- _kuma:dynamicconfig
+- kuma:envoy:admin
+
+Routes (not all of them even have a name):
+- _kuma:dynamicconfig:/dns
+- 9Zuf5Tg79OuZcQITwBbQykxAk2u4fRKrwYn3//AL4Yo= (default route)
+
+#### Other meshes
+
+##### Istio
+
+Istio doesn't seem to be following any naming convention:
+- clusters
+  - some are in form of [inbound|outbound]|port (like `outbound|9080`)
+  - PassthroughCluster / InboundPassthroughCluster
+  - BlackHoleCluster
+  - prometheus_stats
+  - sds-grpc
+  - xds-grpc
+  - agent
+- listeners
+  - ip_port
+  - virtualOutbound
+  - virtualInbound
+
+## Use cases
+
+### Looking at a resource name I want to easily relate it to Kuma concepts
+
+### As a user I want my MeshProxyPatch to continue working after the change
+
+I don't think that's going to be possible.
+In MeshProxyPatch you can use `name` matcher to match a resource name but if we change the name of the resource, it will not match anymore.
+
+### As a user I want to easily exclude all stats related to internal resources from a query
+
+So instead 
+
+``` 
+sum:envoy.cluster.upstream_rq.count{!envoy_cluster:kuma_readiness , !envoy_cluster:access_log_sink , !envoy_cluster:meshtrace_datadog}.as_count()
+```
+
+we can do:
+
+``` 
+sum:envoy.cluster.upstream_rq.count{!envoy_cluster:_kuma*}.as_count()
+```
+
+**Is this a valid use case if we have profiles in MeshMetric?**
+
+### As an operator I want to drop all stats related to internal resources
+
+So I can use a processor like:
+
+```yaml
+processors:
+  filter/drop_kuma_internal_clusters:
+    metrics:
+      include:
+        match_type: regexp
+        metric_names: [".*"]  # apply to all metrics
+        attributes:
+          - key: envoy_cluster
+            value: "^_.*"  # matches if envoy_cluster starts with _
+```
+
+**Is this a valid use case if we have profiles in MeshMetric?**
```

Comment:
I think with profiles you're just shifting the complexity of maintaining the list of internal resources to a different place. So it's still a valid use case